### PR TITLE
Move seed from utils to training

### PR DESCRIPTION
**Pull request by RdoubleA**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Addresses #1441.

#### Changelog
What are the changes made in this PR?

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.)

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Example of docstring: https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285
Example in our docs: https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models

- [ ] I did not change any public API;
- [ ] I have added an example to docs or docstrings;

### CODE REVIEW

File: docs/source/api_ref_training.rst [hunk 1]
```diff
@@ -38,6 +38,16 @@ Utilities for working in a reduced precision setting.
...
 
+
+Miscellaneous
+-------------
+
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+
+    set_seed
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_training.rst:49]: Why does this need to be a public API?

**Review by SalmanMohammadi — APPROVED** [review]

**Review by joecummings — CHANGES_REQUESTED** [review]
This may be one of those rare cases in which it belongs actually in a common utils, rather than training.

My reasoning here is that we need this set_seed for generation and eval, as well, which are not training. Therefore, it would be confusing to import from trainiing.

**Comment by RdoubleA** [comment]
> My reasoning here is that we need this set_seed for generation and eval, as well, which are not training. Therefore, it would be confusing to import from trainiing.

A few of the utils APIs fall under this category, like set_device, set_dtype, etc. But I don't have strong opinions either way, if we leave set_seed in utils then I can close this PR.

**Comment by joecummings** [comment]
> > My reasoning here is that we need this set_seed for generation and eval, as well, which are not training. Therefore, it would be confusing to import from trainiing.
> 
> A few of the utils APIs fall under this category, like set_device, set_dtype, etc. But I don't have strong opinions either way, if we leave set_seed in utils then I can close this PR.

Yeah I'd prefer to leave those in utils or rename to common or something. not sure.

**Comment by SalmanMohammadi** [comment]
There's `torchune.modules.common_utils.py`

**Comment by joecummings** [comment]
> There's `torchune.modules.common_utils.py`

Nah, I don't think these should live nested in modules/. Let's keep it where it is for now and we can discuss if we need a more descriptive name than utils.

**Review by joecummings — APPROVED** [review]
not happy about it, but unblocking for now :/
